src/main/java/pl/betse/beontime/timesheet/controller/MonthTimeEntryController.java

```diff
@@ -0,0 +1,89 @@
+    }
+
+    @GetMapping("/{managerGuid}/consultant/{userGuid}/month/{monthNumber}")
+    public ResponseEntity<Resources<MonthTimeEntryBody>> getMonthForUser(@PathVariable("managerGuid") String managerGuid,
+                                                                         @PathVariable("userGuid") String userGuid,
+                                                                         @PathVariable("monthNumber") String month) {
+        List<List<TimeEntryBo>> timeEntryBoList = timeEntryService.findByUserAndMonth(userGuid, prepareRequestDateForService(month));
+        List<MonthTimeEntryBody> monthTimeEntryBodyList = new ArrayList<>();
+        timeEntryBoList.forEach(monthEntry ->
+                monthTimeEntryBodyList.add(timeEntryMapper.fromTimeEntryBoToMonthTimeEntryBody(monthEntry))
+        );
+        URI location = linkTo(methodOn(MonthTimeEntryController.class).getMonthForUser(managerGuid, userGuid, month)).toUri();
+        Link link = new Link(location.toString(), "self");
+        return ResponseEntity.ok(new Resources<>(monthTimeEntryBodyList, link));
+    }
+
+    @PutMapping("/{managerGuid}/consultant/{userGuid}/month/{monthNumber}")
+    public ResponseEntity editMonthForUser(@RequestBody @Validated(CreateTimeEntry.class) MonthTimeEntryBody monthTimeEntryBody,
+                                           @PathVariable("managerGuid") String managerGuid,
+                                           @PathVariable("userGuid") String userGuid,
+                                           @PathVariable("monthNumber") String month,
+                                           HttpServletRequest httpServletRequest) {
+        LocalDate requestedDate = prepareRequestDateForService(month);
+        List<TimeEntryBo> timeEntryBoList = prepareDataForService(monthTimeEntryBody);
+        timeEntryService.checkIfDateHasCorrectMonth(timeEntryBoList, requestedDate);
+        timeEntryService.checkIfTimeEntriesExist(timeEntryBoList, httpServletRequest.getMethod());
+        timeEntryService.verifyStatusesBeforeAddingComment(timeEntryBoList);
+        timeEntryService.editMonthStatusesAndComments(timeEntryBoList, userGuid, requestedDate);
+        return ResponseEntity.ok().build();
+    }
+
+    private List<TimeEntryBo> prepareDataForService(MonthTimeEntryBody monthTimeEntryBody) {
+        return monthTimeEntryBody.getMonthDays().stream()
+                .map(entry -> timeEntryMapper.fromMonthDayBodyToBo(entry, monthTimeEntryBody))
+                .collect(Collectors.toList());
+    }
+
+    private void checkMonthNumberFormat(String month) {
+        if (!month.matches("[1-3]\\d{3}-[0-1]\\d-01")) {
+            log.error("Incorrect month format or number");
+            throw new IncorrectMonthFormatException();
+        }
+    }
+
+    private LocalDate prepareRequestDateForService(String month) {
+        month += "-01";
+        checkMonthNumberFormat(month);
+        return LocalDate.parse(month);
```

Comment:
I don't understand that logic - looks weird to me

Reply:
We try to resolve problem in that way, because FE app will send to us month in format: YYYY-MM, and we need to change that 'incomplete string' for LocalDate parser, so we added that 'weird logic'.